lib/ruby_indexer/lib/ruby_indexer/reference_finder.rb

```diff
@@ -341,17 +327,27 @@ def collect_constant_references(name, location)
       entries = @index.resolve(name, @stack)
       return unless entries
 
-      previous_reference = @references.last
-
-      entries.each do |entry|
-        next unless entry.name == @target.fully_qualified_name
+      # Filter down to all constant declarations that match the expected name and type
+      matching_entries = entries.select do |e|
+        [
+          Entry::Namespace,
+          Entry::Constant,
+          Entry::ConstantAlias,
+          Entry::UnresolvedConstantAlias,
+          Entry::Class
+        ].include?(e.class) &&
```

Comment:
Sorry, I just realized we should be using `is_a?` since some of these entries inherit from a common parent class. Then you can get rid of `Class` since that is a `Namespace`.

```suggestion
        [
          Entry::Namespace,
          Entry::Constant,
          Entry::ConstantAlias,
          Entry::UnresolvedConstantAlias,
        ].any? { |klass| e.is_a?(klass) } &&
```